test/test_magnet.cpp

```diff
@@ -224,6 +224,23 @@ TORRENT_TEST(magnet)
 	p2 = s->abort();
 }
 
+TORRENT_TEST(parse_escaped_hash_parameter)
+{
+	error_code ec;
+	add_torrent_params p;
+	parse_magnet_uri("magnet:?xt=urn%3Abtih%3Acdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcd", p, ec);
```

Comment:
I wrote a Python script which generates magnet links. Those links don't work with libtorrent right now because everything gets escaped.

```python
>>> from urllib import urlencode
>>> urlencode({'xt': 'urn:btih:1234'})
'xt=urn%3Abtih%3A1234'
```

After re-reading [RFC 3986](https://tools.ietf.org/html/rfc3986#section-3.3) it seems that `:` is a valid character in the query part and does not need to be escaped:

```
query       = *( pchar / "/" / "?" )
pchar       = unreserved / pct-encoded / sub-delims / ":" / "@"
unreserved  = ALPHA / DIGIT / "-" / "." / "_" / "~"
sub-delims  = "!" / "$" / "&" / "'" / "(" / ")" / "*" / "+" / "," / ";" / "="
```

I think the implementation should still handle the case when it is escaped, though.